aws/resource_aws_ram_resource_share_accepter.go

```diff
@@ -0,0 +1,297 @@
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsRamResourceShareAccepterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	in := &ram.AcceptResourceShareInvitationInput{
+		ClientToken: aws.String(resource.UniqueId()),
+	}
+
+	if v, ok := d.GetOk("invitation_arn"); ok {
+		in.ResourceShareInvitationArn = aws.String(v.(string))
+	} else if v, ok := d.GetOk("share_arn"); ok {
+		// need to find invitation arn
+		invitationARN, err := resourceAwsRamResourceShareGetInvitationARN(conn, v.(string))
+		if err != nil {
+			return err
+		}
+
+		in.ResourceShareInvitationArn = aws.String(invitationARN)
+	} else {
+		return fmt.Errorf("Either an invitation ARN or share ARN are required")
+	}
+
+	log.Printf("[DEBUG] Accept RAM resource share invitation request: %s", in)
+	out, err := conn.AcceptResourceShareInvitation(in)
+	if err != nil {
+		return fmt.Errorf("Error accepting RAM resource share invitation: %s", err)
+	}
+
+	d.SetId(aws.StringValue(out.ResourceShareInvitation.ResourceShareInvitationArn))
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ram.ResourceShareInvitationStatusPending},
+		Target:  []string{ram.ResourceShareInvitationStatusAccepted},
+		Refresh: resourceAwsRamResourceShareAccepterStateRefreshFunc(conn, d.Id()),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RAM resource share (%s) state: %s", d.Id(), err)
+	}
+
+	return resourceAwsRamResourceShareAccepterRead(d, meta)
+}
+
+func resourceAwsRamResourceShareAccepterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	request := &ram.GetResourceShareInvitationsInput{
+		ResourceShareInvitationArns: []*string{aws.String(d.Id())},
+	}
+
+	out, err := conn.GetResourceShareInvitations(request)
+	if err != nil {
+		if isAWSErr(err, ram.ErrCodeUnknownResourceException, "") {
+			log.Printf("[WARN] No RAM resource share invitation by ARN (%s) found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading RAM resource share invitation %s: %s", d.Id(), err)
+	}
+
+	if len(out.ResourceShareInvitations) == 0 {
+		log.Printf("[WARN] No RAM resource share invitation by ARN (%s) found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	invitation := out.ResourceShareInvitations[0]
+
+	d.Set("status", invitation.Status)
+	d.Set("receiver_account_id", invitation.ReceiverAccountId)
+	d.Set("sender_account_id", invitation.SenderAccountId)
+	d.Set("share_arn", invitation.ResourceShareArn)
+	d.Set("invitation_arn", invitation.ResourceShareInvitationArn)
+	d.Set("share_id", resourceAwsRamResourceShareGetIDFromARN(aws.StringValue(invitation.ResourceShareArn)))
+	d.Set("share_name", invitation.ResourceShareName)
+
+	var nextToken string
```

Comment:
Same here, we should prefer the paginated AWS Go SDK function [`ListResourcesPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/ram/#RAM.ListResourcesPages) here 👍 

```go
	listResourcesInput := &ram.ListResourcesInput{
		ResourceOwner:     aws.String(ram.ResourceOwnerOtherAccounts),
		ResourceShareArns: []*string{invitation.ResourceShareArn)},
		Principal:         invitation.SenderAccountId,
	}
	var resourceARNs []string

	err = conn.ListResourcesPages(listResourcesInput, func(page *ram.ListResourcesOutput, lastPage bool) bool {
		for _, r := range page.Resources {
			resourceARNs = append(resourceARNs, aws.StringValue(resource.Arn))
		}
		
		return !lastPage
	})

	if err != nil {
		return fmt.Errorf("error listing resources for RAM Resource Share (%s): %s", d.Id(), err)
	}
	
	if err := d.Set("resources", resourceARNs); err != nil {
		return fmt.Errorf("unable to set resources: %s", err)
	}
```